src/components/CodePreview.tsx

```diff
@@ -14,6 +16,11 @@ interface CodePreviewProps {
 
 export function CodePreview({ code, raw }: CodePreviewProps) {
   const [hasCopiedToClipboard, setCopiedToClipboard] = useState(false)
+  const [isExplorerOpen] = useAtom(isExplorerOpenAtom)
+
+  const showButtonCopyToCliboard = raw && !isExplorerOpen
+  const notCopied = !hasCopiedToClipboard && !isExplorerOpen
+  const hasCopied = hasCopiedToClipboard && !isExplorerOpen
```

Comment:
I think this could make more sense.

If `isExplorerOpen` is `false`, then the button will not be visible, so we won't have possibly truthy values on `notCopied` neither `hasCopied`, so I think we could refactor this only to:

```suggestion
  const showCopyToClipboardButton = raw && !isExplorerOpen
```

And then, you can use only the `hasCopiedToClipboard` variable to determine the `hasCopied` or `notCopied` values, am I right? I think it's a little bit redundant this way.

Reply:
Actually, I haven't stopped to think about it.